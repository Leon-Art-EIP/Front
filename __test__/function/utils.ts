import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
