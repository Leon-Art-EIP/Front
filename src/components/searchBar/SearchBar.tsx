import SearchIcon from "@mui/icons-material/Search";

/* c8 ignore start */
export interface SearchBarProps {
  onSearch: (query: string) => void;
}

export function SearchBar({ onSearch }: SearchBarProps): JSX.Element {
  return (
    <div className="w-full">
      <div className="relative flex items-center w-full h-12 rounded-3xl focus-within:shadow-lg bg-secondary overflow-hidden">
        <input
          className="peer h-full w-full outline-none text-md placeholder-gray-600 pl-6"
          type="text"
          id="search"
          placeholder="Rechercher ..."
          onChange={(e) => onSearch(e.target.value)}
        />
        <div className="grid place-items-center h-full w-12 text-gray-300">
          <SearchIcon></SearchIcon>
        </div>
      </div>
    </div>
  );
}
/* c8 ignore end */
