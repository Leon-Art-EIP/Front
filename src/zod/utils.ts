export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
