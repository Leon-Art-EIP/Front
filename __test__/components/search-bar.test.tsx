import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import { SearchBar } from "../../src/components/searchBar/SearchBar";

test("SearchBar snapshot", () => {
  testSnapshot(<SearchBar onSearch={() => {}} />);
});
