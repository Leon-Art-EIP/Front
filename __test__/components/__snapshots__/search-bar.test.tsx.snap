// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar snapshot 1`] = `
<div
  className="w-full"
>
  <div
    className="relative flex items-center w-full h-12 rounded-3xl focus-within:shadow-lg bg-secondary overflow-hidden"
  >
    <input
      className="peer h-full w-full outline-none text-md placeholder:text-tertiary placeholder:opacity-50 pl-6"
      id="search"
      onChange={[Function]}
      placeholder="Rechercher ..."
      type="text"
    />
    <div
      className="grid place-items-center h-full w-12 text-tertiary cursor-pointer"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </div>
</div>
`;
